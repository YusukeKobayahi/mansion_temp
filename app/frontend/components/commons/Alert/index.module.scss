@import "~/styles/variables";
@import "~/styles/utils";

.container {
  padding: 30px 30px;
  text-align: center;
  font-size: 18px;
  font-weight: bold;
}

.children {
  margin-top: 30px;
  text-align: center;
}

$brown: $color-main;
$red: #ff3300;

@mixin fill($color: $red) {
  border: 2px solid $color;
  background-color: $color;
  color: white;
}
@mixin outline($color: $red) {
  &.outline {
    border: 2px solid $color;
    background-color: unset;
    color: $color;
  }
}

.red {
  @include fill;
  @include outline;
}

.brown {
  @include fill($brown);
  @include outline($brown);
}
