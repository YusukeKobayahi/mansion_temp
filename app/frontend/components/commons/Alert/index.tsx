import styles from "./index.module.scss";
import classnames from "classnames";

type AlertProps = {
  text: string;
  children?: React.ReactNode;
  className?: string;
  color?: "brown" | "red";
  outline?: boolean;
};
const Alert: React.FC<AlertProps> = ({
  text,
  children,
  className,
  color = "red",
  outline = false,
}: AlertProps) => (
  <div
    className={classnames(
      styles.container,
      styles[color],
      { [styles.outline]: outline },
      className
    )}
  >
    <p>{text}</p>
    {children && <div className={styles.children}>{children}</div>}
  </div>
);

export default Alert;
