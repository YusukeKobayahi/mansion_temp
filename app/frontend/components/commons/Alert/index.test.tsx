import { render } from "@testing-library/react";
import Alert from ".";

describe("components/commons/Alert", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(
      <Alert
        text="条件に合う物件がありません。条件を変更して再度検索してください。"
        outline={true}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
