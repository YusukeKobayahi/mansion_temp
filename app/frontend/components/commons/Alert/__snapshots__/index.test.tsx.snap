// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/Alert 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="container red outline"
  >
    <p>
      条件に合う物件がありません。条件を変更して再度検索してください。
    </p>
  </div>
</DocumentFragment>
`;
